fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -196,6 +197,12 @@ void FixedLagSmoother::optimizationLoop()
       //         queue and obtaining the lock for the graph. But we have now obtained two different locks. If we are
       //         not extremely careful, we could get a deadlock.
       processQueue(*new_transaction);
+      // Skip this optimization cycle if the transaction is empty because something failed while processing the pending
```

Comment:
@svwilliams I've added this as a tentative way to abort running the optimization cycle if the transaction is empty.

It looks like most of the methods handle empty transactions gracefully, except for the graph `update` and `optimize` methods, just looking at the code though.